dev-haskell/hakyll/hakyll-4.16.1.0.ebuild

```diff
@@ -0,0 +1,78 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# ebuild generated by hackport 0.8.4.0.9999
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour test-suite"
+inherit haskell-cabal
+
+DESCRIPTION="A static website compiler library"
+HOMEPAGE="https://jaspervdj.be/hakyll"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+IUSE="buildwebsite +checkexternal +previewserver +usepandoc +watchserver"
+
+RDEPEND=">=dev-haskell/blaze-html-0.5:=[profile?] <dev-haskell/blaze-html-0.10:=[profile?]
+	>=dev-haskell/blaze-markup-0.5.1:=[profile?] <dev-haskell/blaze-markup-0.9:=[profile?]
+	>=dev-haskell/data-default-0.4:=[profile?] <dev-haskell/data-default-0.8:=[profile?]
+	>=dev-haskell/file-embed-0.0.10.1:=[profile?] <dev-haskell/file-embed-0.0.16:=[profile?]
+	>=dev-haskell/hashable-1.0:=[profile?] <dev-haskell/hashable-2:=[profile?]
+	>=dev-haskell/lrucache-1.1.1:=[profile?] <dev-haskell/lrucache-1.3:=[profile?]
+	>=dev-haskell/network-uri-2.6:=[profile?] <dev-haskell/network-uri-2.7:=[profile?]
+	>=dev-haskell/optparse-applicative-0.12:=[profile?] <dev-haskell/optparse-applicative-0.19:=[profile?]
+	>=dev-haskell/parsec-3.0:=[profile?] <dev-haskell/parsec-3.2:=[profile?]
+	>=dev-haskell/random-1.0:=[profile?] <dev-haskell/random-1.3:=[profile?]
+	>=dev-haskell/regex-tdfa-1.1:=[profile?] <dev-haskell/regex-tdfa-1.4:=[profile?]
+	>=dev-haskell/resourcet-1.1:=[profile?] <dev-haskell/resourcet-1.4:=[profile?]
+	>=dev-haskell/scientific-0.3.4:=[profile?] <dev-haskell/scientific-0.4:=[profile?]
+	>=dev-haskell/tagsoup-0.13.1:=[profile?] <dev-haskell/tagsoup-0.15:=[profile?]
+	>=dev-haskell/time-locale-compat-0.1:=[profile?] <dev-haskell/time-locale-compat-0.2:=[profile?]
+	>=dev-haskell/unordered-containers-0.2:=[profile?] <dev-haskell/unordered-containers-0.3:=[profile?]
+	>=dev-haskell/vector-0.11:=[profile?] <dev-haskell/vector-0.14:=[profile?]
+	>=dev-haskell/wai-app-static-3.1:=[profile?] <dev-haskell/wai-app-static-3.2:=[profile?]
+	>=dev-haskell/yaml-0.8.11:=[profile?] <dev-haskell/yaml-0.12:=[profile?]
+	>=dev-lang/ghc-8.10.6:=
+	|| ( ( >=dev-haskell/aeson-1.0:=[profile?] <dev-haskell/aeson-1.6:=[profile?] )
```

Comment:
is it possible to tweak hackport to do it?

Reply:
It should be possible. I just haven't had success at doing it quickly in the past, so it hasn't happened. We need to interpret `|| ^>= <package> || ^>= <package>` in the .cabal file differently, such that we collapse all intermediate version bounds made redundant by an overarching lower and upper bound. I have never seen `^>=` used in any of the thousands of packages we maintain where it explicitly skips a major version, so it should be safe to default to taking the lowest bound and the highest bound.

There may even be cabal logic already within the cabal library which can transpose these version bounds for us.

Reply:
@solpeth I just wanted to point out some relevant bugs in case you end up working on this:

https://github.com/gentoo-haskell/hackport/issues/58
https://github.com/gentoo-haskell/hackport/issues/92 (loosely relevant)
https://github.com/gentoo-haskell/hackport/issues/120
https://github.com/gentoo-haskell/hackport/issues/127 (dupe of https://github.com/gentoo-haskell/hackport/issues/58)